lib/Doctrine/DBAL/Platforms/Keywords/MariaDb12Keywords.php

```diff
@@ -0,0 +1,280 @@
+<?php
+/*
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\DBAL\Platforms\Keywords;
+
+/**
+ * MariaDb reserved keywords list.
+ *
+ */
+class MariaDb12Keywords extends MySQLKeywords
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
```

Comment:
Return type

Reply:
Good, still have a question. typehint or phpdoc style ? Do you intend to merge it back to 2.5 series ? I'm using 7.1 so no problem, but I think this fix should be in 2.5.x also (the P/R not only fix mariadb 10.2 but also detection of 5.5.5-10.x.x versions).

Reply:
I'll typehint for now

Reply:
@belgattitude typehint - this patch will go only into `2.6`, as it is not a security issue.